backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/migration/MigrationTask.java

```diff
@@ -148,7 +157,7 @@ public TaskType type() {
 
     @Override
     public Optional<TaskExecutionDetails.AdditionalInformation> details() {
-        return Optional.of(new AdditionalInformations(target));
+        return Optional.of(new AdditionalInformations(target, Clock.systemUTC().instant()));
```

Comment:
This 'now' timestamp seems like it is repeated everywhere. Couldn't we make it a tld field like submitDate to avoid massive repetition?

Reply:
it must be part of the AdditionalInformation because its related to it, not to an event

Reply:
Can we think of a mechanism to stay "dry"?

Reply:
The timestamp is in the parent class, it's not repeated in the class hierarchy.
However, the point is, when you create an AdditionalInformation derivative, you want the to store the associated timestamp. As every AdditionalInformation derivative have to pass it to its parent, I don't see any choice but to pass it in the constructor.
The alternative is to create a Factory for every types where you could provide the Clock and then creating a new instance would not require to provide the timestamp.
It would be even more verbose.
The changes are massive because we have a massive amount of different classes.
If we really want to tackle this problem, we could actually say that AdditionalInformation could contain just a Map. Because we probably never use the types (I may be wrong about that).
We can debate that further, every step to minimize the codebase is worth it, but I don't see that as something specific of this commit.